libqpdf/QPDF.cc

```diff
@@ -2461,6 +2461,10 @@ QPDF::getRoot()
     QPDFObjectHandle root = this->m->trailer.getKey("/Root");
     if (!root.isDictionary()) {
         throw damagedPDF("", 0, "unable to find /Root dictionary");
+    } else if (
+        m->check_mode && !root.getKey("/Type").isNameAndEquals("/Catalog")) {
```

Comment:
Can you add a comment here explaining that check_mode is an interim step toward a more comprehensive concept of warning levels? Or something like that? I just don't want to forget about it if we ever do go through the exercise of looking at all warnings. Feel free to reference this PR in the comment. If we were going to touch all warnings, there are several things people have asked for such as a table somewhere where someone could look up a warning and get a more detailed explanation. Whenever I think about this, I'm also painfully aware of the fact that the qpdf source code is not structured for internationalization either....but that's another whole issue and out of scope for this.

Reply:
Done.